src/environmentd/src/deployment/preflight.rs

```diff
@@ -237,25 +241,76 @@ async fn check_ddl_changes(
     initial_next_user_item_id: u64,
     initial_next_replica_id: u64,
 ) {
-    let (next_user_item_id, next_replica_id) = get_next_ids(
-        boot_ts,
-        persist_client.clone(),
-        environment_id.clone(),
-        deploy_generation,
-        Arc::clone(&catalog_metrics),
-        bootstrap_args.clone(),
+    let openable_adapter_storage = mz_catalog::durable::persist_backed_catalog_state(
+        persist_client,
+        environment_id.organization_id(),
+        BUILD_INFO.semver_version(),
+        Some(deploy_generation),
+        catalog_metrics,
     )
-    .await;
+    .await
+    .expect("incompatible catalog/persist version");
+
+    let (mut catalog, _audit_logs) = openable_adapter_storage
+        .open_savepoint(boot_ts, &bootstrap_args)
+        .await
+        .unwrap_or_terminate("can open in savepoint mode");
+
+    let tx = catalog
+        .transaction()
+        .await
+        .unwrap_or_terminate("unexpected error while getting transaction");
+
+    // We must explicitly check the catalog for these IDs since IDs can be
+    // allocated during sequencing/planning but not yet committed to the catalog.
```

Comment:
Maybe good to spell out that only may the objects with these IDs have not _yet_ been committed to the catalog, but they might never be because their sequencing has been aborted.